### PR TITLE
ref(slack): Account for different commit URL formats

### PR DESCRIPTION
**Pull request by isabellaenriquez**
Different providers have different commit URL formats, so this PR makes sure we render the correct commit URL in Slack issue alerts with suspect commits depending on the provider the commit is tied to.

### CODE REVIEW

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 1]
```diff
@@ -58,6 +58,18 @@
...
+    "github": "{base_url}/commit/{commit_id}",
+    "integrations:github": "{base_url}/commit/{commit_id}",
+    "integrations:github_enterprise": "{base_url}/commit/{commit_id}",
+    "visualstudio": "{base_url}/commit/{commit_id}",
+    "integrations:vsts": "{base_url}/commit/{commit_id}",
+    "gitlab": "{base_url}/commit/{commit_id}",
+    "integrations:gitlab": "{base_url}/commit/{commit_id}",
+    "bitbucket": "{base_url}/commits/{commit_id}",
+    "integrations:bitbucket": "{base_url}/commits/{commit_id}",
+}
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/issues.py:71]: ill admit there might be a better way of tracking this 😬

File: tests/sentry/integrations/slack/test_message_builder.py [hunk 2]
```diff
@@ -467,6 +467,93 @@ def test_issue_alert_with_suspect_commits(self, mock_external_url):
...
+            group=group,
+            event=event,
+            suspect_commit=suspect_commit,
+        )
+
+    @patch(
+        "sentry.api.serializers.models.pullrequest.PullRequestSerializer._external_url",
+        return_value="https://unknown.com/meowmeow/cats/pull/1",
+    )
+    @with_feature("organizations:slack-block-kit")
```
Comment (isabellaenriquez) [inline tests/sentry/integrations/slack/test_message_builder.py:547]: would it be worth manually setting what we expect for the suspect commits block? right now these tests use the function in the message builder that builds it (`get_suspect_commit_text`)

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 3]
```diff
@@ -58,6 +58,18 @@
...
 
 STATUSES = {"resolved": "resolved", "ignored": "ignored", "unresolved": "re-opened"}
+PROVIDER_TO_COMMIT_LINK_URL_FORMAT = {
+    "github": "{base_url}/commit/{commit_id}",
+    "integrations:github": "{base_url}/commit/{commit_id}",
+    "integrations:github_enterprise": "{base_url}/commit/{commit_id}",
+    "visualstudio": "{base_url}/commit/{commit_id}",  # TODO: buggy
+    "integrations:vsts": "{base_url}/commit/{commit_id}",
+    "gitlab": "{base_url}/commit/{commit_id}",  # TODO: buggy
+    "integrations:gitlab": "{base_url}/commit/{commit_id}",  # TODO: buggy
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/issues.py:68]: these three provider ids i get serialization errors if i use them in tests, are these outdated? this list of providers is based off that of `commitLink.tsx`

File: tests/sentry/integrations/slack/test_message_builder.py [hunk 4]
```diff
@@ -608,40 +663,45 @@ def test_issue_alert_with_suggested_assignees(self):
 
         # auto assign group
         ProjectOwnership.handle_auto_assignment(self.project.id, event)
-        suspect_commit = {
-            "commit_id": commit.key,
-            "author_email": commit.author.email,
-        }
+        suspect_commit_text = f"Suspect Commit: {commit.key[:6]} by {user2.email}"  # no commit link because there is no PR
```
Comment (isabellaenriquez) [inline tests/sentry/integrations/slack/test_message_builder.py:666]: we can get the repo through the commit as well; would it be worth linking to the commit even if there is no PR?

(probably something that can be implemented in a separate PR)

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 5]
```diff
@@ -58,6 +58,16 @@
...
 
 STATUSES = {"resolved": "resolved", "ignored": "ignored", "unresolved": "re-opened"}
+SUPPORTED_PROVIDER_TO_COMMIT_URL = {
+    "github": "{base_url}/commit/{commit_id}",
+    "integrations:github": "{base_url}/commit/{commit_id}",
+    "integrations:github_enterprise": "{base_url}/commit/{commit_id}",
+    "integrations:vsts": "{base_url}/commit/{commit_id}",
+    "integrations:gitlab": "{base_url}/commit/{commit_id}",
+    "bitbucket": "{base_url}/commits/{commit_id}",
+    "integrations:bitbucket": "{base_url}/commits/{commit_id}",
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/issues.py:68]: these are based off the repo providers added to `bindings` within `src/sentry/integrations/` and `src/sentry_plugins/`

File: src/sentry/integrations/slack/message_builder/issues.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/integrations/slack/message_builder/issues.py]: What if you removed all the dupes / integration starting ones like:

```
SUPPORTED_PROVIDER_TO_COMMIT_URL = {
    "github": "{base_url}/commit/{commit_id}",
    "vsts": "{base_url}/commit/{commit_id}",
    "gitlab": "{base_url}/commit/{commit_id}",
    "bitbucket": "{base_url}/commits/{commit_id}",
}
```

and then where they're being checked instead of:

```
        if repo_base and provider in SUPPORTED_PROVIDER_TO_COMMIT_URL:
            commit_link = f"<{SUPPORTED_PROVIDER_TO_COMMIT_URL[provider].format(base_url=repo_base, commit_id=commit_id)}|{commit_id[:6]}>"
            suspect_commit_text += f"{commit_link} by {author_display}"
```

you did:

```
if repo_base:
    for supported_provider, url in SUPPORTED_PROVIDER_TO_COMMIT_URL.items():
        if supported_provider in provider:
            commit_link = f"<{url.format(base_url=repo_base, commit_id=commit_id)}|{commit_id[:6]}>"
```

or since bitbucket is the only different one maybe like:

```
if repo_base:
   if "bitbucket" in provider:
        commit_link = f"{base_url}/commits/{commit_id}"
    else:
        ...
```

File: tests/sentry/integrations/slack/test_message_builder.py (hunk 4 above)
Comment (ceorourke) [reply tests/sentry/integrations/slack/test_message_builder.py:667]: I have no idea if that scenario happens 😅 for now let's just leave it w/o linking it

**Review by ceorourke — APPROVED** [review]
LGTM, feel free to request a re-review if you consolidate the `SUPPORTED_PROVIDER_TO_COMMIT_URL` and want another look.

**Review by ceorourke — APPROVED** [review]
